### PR TITLE
fix: missing pull_requests permission is no longer fatal

### PR DESCRIPTION
**Pull request by KevinDeJong-TomTom**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+### Fixed
+- Missing `pull_requests` permission is no longer fatal when creating a new GitHub Release
```
Comment (maikelvdh) [inline CHANGELOG.md:9]:
```suggestion
- Missing `pull-requests` permission is no longer fatal when creating a new GitHub Release
```

File: dist/bump/index.js [hunk 2]
```diff
@@ -12333,12 +12333,21 @@ exports.getTags = getTags;
...
+                repo: REPO,
+                commit_sha: sha,
+            });
+            return prs;
+        }
+        catch (error) {
+            if (error.message !== "Resource not accessible by integration") {
+                throw error;
+            }
+            core.warning("Could not retrieve Pull Request references, did you forget to enable the `pull_requests` permission?");
```
Comment (maikelvdh) [inline dist/bump/index.js:12348]:
```suggestion
            core.warning("Could not retrieve Pull Request references, did you forget to enable the `pull-requests` permission?");
```

**Review by maikelvdh — CHANGES_REQUESTED** [review]
Change reference to `pull-requests` instead of `pull_requests`

File: dist/bump/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoostMuller-TomTom) [reply dist/bump/index.js:12348]: Also note `pull-requests` instead of `pull_requests`.

**Review by jstmlr — APPROVED** [review]
Pending `s/pull_request/pull-request/` (see Maikel's comment), this will suffice as a containment.
We can look into the API's, and see if there's a better way to check permissions upfront later.

I'll take note to also:
- update the README with info regarding permissions, and
- use a `permissions` block in commisery-action itself, to detect such issues earlier.

File: dist/bump/index.js [hunk 3]
```diff
@@ -12333,12 +12333,21 @@ exports.getTags = getTags;
...
+        try {
+            const { data: prs } = yield getOctokit().rest.repos.listPullRequestsAssociatedWithCommit({
+                owner: OWNER,
+                repo: REPO,
+                commit_sha: sha,
+            });
+            return prs;
+        }
+        catch (error) {
+            if (error.message !== "Resource not accessible by integration") {
```
Comment (MartijnLeijssen-TomTom) [inline dist/bump/index.js:12345]: Not sure if they (Github) made a specific error class for this, if so, I would rather depend on the error class
```
if(error instanceof GHError) {
```
Then this error prone string comparison

Comment (MartijnLeijssen-TomTom) [reply dist/bump/index.js:12345]: Hmm, tried to google it myself, but didn't find it.

Comment (KevinDeJong-TomTom) [reply dist/bump/index.js:12345]: Unfortunately it isn't :(

File: dist/bump/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (KevinDeJong-TomTom) [reply dist/bump/index.js]: removed the warning (as discussed offline with Joost and Maikel) as this is causing spam spam spam.
We will look into improving this after fixing the current issue.

**Pull request merged**